vault/token_store.go

```diff
@@ -1254,12 +1254,16 @@ func (ts *TokenStore) handleCreateCommon(
 	// the client has root or sudo privileges
 	case !isSudo:
 		// Sanitize passed-in and parent policies before comparison
-		sanitizedInputPolicies := policyutil.SanitizePolicies(data.Policies, true)
-		sanitizedParentPolicies := policyutil.SanitizePolicies(parent.Policies, true)
+		sanitizedInputPolicies := policyutil.SanitizePolicies(data.Policies, false)
+		sanitizedParentPolicies := policyutil.SanitizePolicies(parent.Policies, false)
 
 		if !strutil.StrListSubset(sanitizedParentPolicies, sanitizedInputPolicies) {
 			return logical.ErrorResponse("child policies must be subset of parent"), logical.ErrInvalidRequest
 		}
+
+		if !strutil.StrListContains(sanitizedParentPolicies, "default") {
+			data.NoDefaultPolicy = true
```

Comment:
Adding the `default` policy to the tokens created on token roles also sounds like a valid case to me. However, adding it when `allowed_policies` and `disallowed_policies` are set does seem off, as you pointed out. I'll address that.